src/test/ui/issue-49257.rs

```diff
@@ -0,0 +1,23 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// Test for #49257:
+// emits good diagnostics for `..` pattern fragments not in the last position.
+
+#![allow(unused)]
+
+struct Point { x: u8, y: u8 }
+
+fn main() {
+    let p = Point { x: 0, y: 0 };
+    let Point { .., y } = p; //~ ERROR expected `}`, found `,`
+    //~| ERROR pattern does not mention field `x`
+    //~| ERROR pattern does not mention field `y`
```

Comment:
Note that the pattern mentions field `y` actually.
Ideally we should do recovery in the parser and continue parsing fields after encountering incorrect `..`, but it requires somewhat larger refactoring of `fn parse_pat_fields`.

Reply:
@petrochenkov Would it be possible to just hide the "pattern does not metion field" errors?

Reply:
@Badel2 
Yes, but in this case `Pat { .., field }` or `Pat { field, .., }` have a single obvious meaning, so it's better to use recovery at syntax level and starting from that assume that `Pat { field, .. }` was actually implied.